drivers/flash/soc_flash_nrf.c

```diff
@@ -95,6 +99,24 @@ static inline bool is_addr_valid(off_t addr, size_t len)
 	return true;
 }
 
+#ifdef CONFIG_SOC_FLASH_NRF_UICR
+static inline bool is_uicr_addr_valid(off_t addr, size_t len)
```

Comment:
inline is unneeded

Reply:
@pdunaj why not?

Reply:
It is better to leave this for the compiler's optimizer. It will choose what's best.
Compiler will most probably inline it anyway especially that this is small function with only one call to it.

Reply:
`inline` is only a hint; the compiler will make the decision anyway.  GCC may ignore it and do what it wants in all cases, but other compilers are starting to be used and they might benefit from the clue.

The `static inline` idiom is common for this type of function throughout Zephyr.  It's part of the declaration for `is_addr_valid` already in this file and not changed in this PR, and it's used in several of the other flash drivers as well.  IMO the as-submitted code is right to continue to use it.

Reply:
My impression was that in 2018 all compilers used for production code have reasonably good optimization logic or are out of market.
But certainly leaving it will add no harm beside few additional characters being used. I doubt there are any benefits from it but there is no harm.

Reply:
kept as is